src/lib/spinel/radio_spinel.cpp

```diff
@@ -1422,6 +1422,19 @@ otError RadioSpinel::RequestV(uint32_t command, spinel_prop_key_t aKey, const ch
     EXPECT(tid > 0, error = OT_ERROR_BUSY);
 
     error = GetSpinelDriver().SendCommand(command, aKey, tid, aFormat, aArgs);
+
+    if( mTxRadioTid && (error == OT_ERROR_BUSY) )
+    {
+        // Handle RETRY mechanism in case of VALUE_SET / STREAM_RAW failed transaction
+        error = WaitResponse();
+
+        if( error == OT_ERROR_NONE )
+        {
+            // Call again the current command. SPI interface shall be available now !
+            error = mSpinelDriver.SendCommand(command, aKey, tid, aFormat, aArgs);
+        }
+    }
+
```

Comment:
When the host sends `STREAM_RAW ` to the RCP via SPI interface, the `SpiInterface::SendFrame()` calls the `PushPullSpi()` to send the frame to the RCP via SPI bus, and then the `mSpiTxIsReady` will be set to false. If you meet the error `mSpiTxIsReady`, which means the SPI bus failed to send the frame to the RCP. Could you find out why the SPI transmission fails? I think this is the root cause of the issue.

Reply:
SPI driver returns BUSY at first OT_ERROR_BUSY. if SendCommand returns BUSY the function will fail

Reply:
Is it possible to find out why the SPI driver returns BUSY? 

I think even if we retransmit the Spinel Command, it still has a chance to fail if we do not resolve the root cause.

Reply:
![MicrosoftTeams-image (5)](https://github.com/openthread/openthread/assets/104561745/995fff42-81b6-4871-b3e3-163636d1f714)
SendCommand returns BUSY as the driver is still processing the previous one. Driver is expecting a retry from OT host app.

Reply:
In general, the SPI `MISO` and `MOSI` should shift out data at the same time. I saw there is a delay after the `MOSI`  shift out data and then the`MISO` starts to shift out data. Which causes the SPI master receives too many `0xff`from the SPI slave. Then the [SpiInterface](https://github.com/openthread/openthread/blob/main/src/posix/platform/spi_interface.cpp#L486) thoughts that the SPI frame is failed to be sent to the RCP, and then it will re-transmit the frame to the RCP at the next time. If there are other Spinel frames are sent to the RCP at this time point, the [SendFrame()](https://github.com/openthread/openthread/blob/main/src/posix/platform/spi_interface.cpp#L784) will return `BUSY`.  Could you check your SPI slave driver to find out why there is a huge delay between `MOSI` and `MISO` signals.

Reply:
It's normal. A spi header is configured (for MISO) with 16bytes 0xFF at each SPI transaction. It is correctly configured on OT Host as well.

Reply:
The function `GetRealRxFrameStart()` will skip at most 16 bytes `0xff` by default. Your RCP may send more than 16 bytes `0xff` to the host. 

<1> Could you check how many `0xff` are received from the RCP?
<2> Could you change the `mSpiAlignAllowance` using the `spi-align-allowance` option to a larger value to check if the issue still exits?
<3> Could you show the debug level log of the SpiInterface when the issue happens?